Machine Learning/Regression Loss Functions/regression-loss-functions.ipynb

```diff
@@ -0,0 +1,415 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from sympy.stats import LogNormal, density\n",
+    "z = symbols('z')\n",
+    "log_norm_1 = LogNormal(\"x\", 0, 1.0)\n",
+    "log_norm_2 = LogNormal(\"x\", 0, 0.5)\n",
+    "log_norm_3 = LogNormal(\"x\", 0, 0.25)\n",
+    "pdf_plot1= plot(density(log_norm_1)(z), (z,0.01, 5), show=False, line_color=\"blue\")\n",
+    "pdf_plot2= plot(density(log_norm_2)(z), (z,0.01, 5), show=False, line_color=\"green\")\n",
+    "pdf_plot3= plot(density(log_norm_3)(z), (z,0.01, 5), show=False, line_color=\"red\")\n",
+    "pdf_plot1.extend(pdf_plot2)\n",
+    "pdf_plot1.extend(pdf_plot3)\n",
+    "pdf_plot1.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "The distribution of the values has a \"bump\" for smaller values and then a long tail with a small number of very large values. \n",
+    "\n",
+    "Naturally, when we want to predict the target values, we want to allow for an error depending on the magnitude of the target. It’s ok to miss a population of a metropolitan area by 10 000, but for a small town it is unforgivable. In this case MSLE can be more suitable for the regression job than MSE. As the name suggests, MSLE minimizes the quadratic difference between the logarithms of the actual value and the prediction\n",
+    "$$L(a_i, p_i) = (\\log(a_i + t) - \\log(p_i + t))^2$$\n",
+    "\n",
+    "Qualitatively speaking, MSLE tends to lead to lower error for small targets and for higher error on large targets than MSE. This trade-off can be controlled by adjusting an offset parameter $t$ which is set to 1.0 by default. By increasing this parameter we influence the “transition point” at which we go from minimizing quadratic error to minimizing quadratic log error. \n",
```

Comment:
This is all good. I think it is worth saying that in the log regime it is effectively targeting relative error, i.e.:
`log((1 + 1100) / (1 + 1000))^2` is about the same as `log((1 + 11000) / (1 + 10000))^2` so missing a value by 10% is penalised equivalently whether the actual error is 100 or 1000. Contrasting with MSE, the latter case costs 100x more.

Reply:
Added